cpp/src/arrow/csv/parser.cc

```diff
@@ -324,9 +324,29 @@ class BlockParserImpl {
         if (*(end - 1) == '\r') {
           --end;
         }
-        return MismatchingColumns(batch_.num_cols_, num_cols,
```

Comment:
Since the code for handling an invalid row is becoming non-trivial, perhaps move it into a dedicated method?

Reply:
Done